### PR TITLE
Adding foreman-smart_proxy_discovery

### PR DESCRIPTION
**Pull request by lzap**
Builds to follow in a comment

### CODE REVIEW

**Comment by lzap** [comment]
http://koji.katello.org/koji/taskinfo?taskID=207033
http://koji.katello.org/koji/taskinfo?taskID=207035
http://koji.katello.org/koji/taskinfo?taskID=207037

Just realized that jenkins does this for me.... amending few changes. Ready to merge.

**Comment by lzap** [comment]
Rebuild please [test]

**Comment by domcleal** [comment]
Missing both runtime dependencies: rack-contrib, rest-client.

Incidentally, the .gem on rubygems.org isn't tagged in the git repository.

File: rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+BuildRequires: ruby(abi)
+%else
+Requires: ruby(release)
+BuildRequires: ruby(release)
+%endif
+BuildRequires: rubygems-devel
+BuildRequires: ruby(rubygems)
+
+%if 0%{?rhel}
+%else
```
Comment (domcleal) [inline rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: What's this block for?

File: rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+mkdir -p %{buildroot}%{foreman_proxy_settingsd_dir}
+mv %{buildroot}%{gem_instdir}/settings.d/discovery.yml.example \
+   %{buildroot}%{foreman_proxy_settingsd_dir}/discovery.yml
+
+%files
+%dir %{gem_instdir}
+%{gem_libdir}
+%{gem_spec}
+%{foreman_proxy_bundlerd_dir}/discovery.rb
+%{foreman_proxy_settingsd_dir}/discovery.yml
```
Comment (domcleal) [inline rubygem-smart_proxy_discovery/rubygem-smart_proxy_discovery.spec]: %config

**Comment by lzap** [comment]
Bumped to 1.0.1. Had to add tests, remove that unnecessary dependency and test it on real world scenario. Going to setup Jenkins now. @GregSutcliffe your turn! :-)

**Comment by domcleal** [comment]
The .gem is still 1.0.0.

**Comment by lzap** [comment]
Yeah noticed. Fixed.

**Comment by domcleal** [comment]
DEBUG util.py:264:  Error: No Package found for rubygem(rack-test) on el7

**Comment by lzap** [comment]
Yeah removed all the test-related build deps as we don't provide any testing anyway.

**Comment by domcleal** [comment]
Thanks @lzap, building now into nightly plugin repo.

**Pull request merged**